// flash-messages-angular/module/module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var flash_messages_component_1 = require("./flash-messages.component");
var flash_messages_service_1 = require("./flash-messages.service");
var FlashMessagesModule = /** @class */ (function () {
    function FlashMessagesModule() {
    }
    FlashMessagesModule_1 = FlashMessagesModule;
    FlashMessagesModule.forRoot = function () {
        return {
            ngModule: FlashMessagesModule_1,
            providers: [flash_messages_service_1.FlashMessagesService]
        };
    };
    var FlashMessagesModule_1;
    FlashMessagesModule = FlashMessagesModule_1 = __decorate([
        core_1.NgModule({
            imports: [common_1.CommonModule],
            declarations: [flash_messages_component_1.FlashMessagesComponent],
            exports: [flash_messages_component_1.FlashMessagesComponent],
            providers: []
        })
    ], FlashMessagesModule);
    return FlashMessagesModule;
}());
exports.FlashMessagesModule = FlashMessagesModule;
