// flash-messages-angular/__ivy_ngcc__/module/module.js
"use strict";
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./flash-messages.component');
var ɵngcc2 = require('@angular/common');
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var flash_messages_component_1 = require("./flash-messages.component");
var flash_messages_service_1 = require("./flash-messages.service");
var FlashMessagesModule = /** @class */ (function () {
    function FlashMessagesModule() {
    }
    FlashMessagesModule_1 = FlashMessagesModule;
    FlashMessagesModule.forRoot = function () {
        return {
            ngModule: FlashMessagesModule_1,
            providers: [flash_messages_service_1.FlashMessagesService]
        };
    };
    var FlashMessagesModule_1;
FlashMessagesModule.ɵfac = function FlashMessagesModule_Factory(t) { return new (t || FlashMessagesModule)(); };
FlashMessagesModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: FlashMessagesModule });
FlashMessagesModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ providers: [], imports: [[common_1.CommonModule]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FlashMessagesModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [common_1.CommonModule],
                declarations: [flash_messages_component_1.FlashMessagesComponent],
                exports: [flash_messages_component_1.FlashMessagesComponent],
                providers: []
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FlashMessagesModule, { declarations: [ɵngcc1.FlashMessagesComponent], imports: [ɵngcc2.CommonModule], exports: [ɵngcc1.FlashMessagesComponent] }); })();
    return FlashMessagesModule;
}());
exports.FlashMessagesModule = FlashMessagesModule;
